src/irmin/commit.ml

```diff
@@ -84,13 +116,15 @@ struct
     | None -> N.add n N.Val.empty
     | Some node -> Lwt.return node
 
+  let equal_key = Type.(unstage (equal Key.t))
   let equal_opt_keys = Type.(unstage (equal (option Key.t)))
 
   let merge_commit info t ~old k1 k2 =
+    Log.debug (fun l -> l "Commit.merge %a %a" pp_key k1 pp_key k2);
     let* v1 = get t k1 in
     let* v2 = get t k2 in
-    if List.mem k1 (Val.parents v2) then Merge.ok k2
```

Comment:
that's a bug in the current implementation, could you open a separate issue for this? (I though we removed all polymorphic equalities from our code 😱 )

Reply:
Done in https://github.com/mirage/irmin/pull/1518.